### PR TITLE
Add Archiecture Decisions Record for Knative based Kyma Event Bus

### PR DESCRIPTION
**Pull request by Abd4llA**
**Description**

Add Archiecture Decisions Record for Knative based Kyma Event Bus

### CODE REVIEW

File: architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+2. Provide `NATS Streaming` operator that provides `NATS Streaming` clusters which satisfies all production needs.
+3. Implementing an `Azure Service Bus Messaging` knative bus implementation as a plan B.
+4. Contribute both implementations to `knative eventing`.
+
+## Status
+
+In Progress
+
+## Consequences
+
```
Comment (klaudiagrz) [inline architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md]: Add an introducing sentence before the list:
`These are the consequences of this architecture decision:`

File: architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+- Refactor `Event Bus Publish` application
```
Comment (klaudiagrz) [inline architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md]: Add dot at the end.

File: architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+- Merge `Event Bus Push` & `Sub Validator` apps into a single `Event Bus Controller` app.
```
Comment (klaudiagrz) [inline architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md]: ``Merge Event Bus Push and Sub Validator applications into a single Event Bus Controller application.``

File: architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+              |                                                 |
+              +-------------------------------------------------+
+
+### NATS Streaming
+
+Since knative provides an abstraction of "The Bus" which is a pluggable component of knative eventing that have different  implementations backed up by a certain messaging store "e.g. Kafka Bus, GCP PubSub Bus, etc..".
+
+### Decision
+
+1. Implementing a production ready `NATS Streaming` knative bus implementation.
```
Comment (klaudiagrz) [inline architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md]: ``1. Implement a production-ready Knative Bus `NATS Streaming`.``
``2. Provide `NATS Streaming` operator with `NATS Streaming` clusters which satisfy all production needs.`` -> not sure whether it's technically correct, though 🤷‍♀️ 
``3. Implement Knative Bus `Azure Service Bus Messaging` as a plan B.``
`4. Contribute both implementations to Knative Eventing.`

**Comment by Abd4llA** [comment]
@klaudiagrz  Hi Kauldia, @radufa applied all the comments, can you check?

File: architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# ADR 018: Knative-based Kyma Eventing 
+
+Created on 2018-08-19 by Ahmed Abdalla (@Abd4llA).
+
+## Context
+
+Based on the decision of building Kyma on top of Knative and along the effort to reuse as much of knative functionality as possible, the eventing area shines as a good area for alignment. That is why a decision was made to have Kyma leverage as much of knative eventing as possible.This is a an architecture decision on how to achieve that target.
```
Comment (klaudiagrz) [inline architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md]: 1. `Based on the decision of building Kyma on top of Knative, and` ...
2. Capitalize Knative and Knative Eventing
3. as possible.This <- space needed here

File: architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+The decisions are to:
+
+1. Implement a production-ready Knative Bus `NATS Streaming`.
+2. Provide `NATS Streaming` operator that provides `NATS Streaming` clusters which satisfy all production needs.
+3. Implement Knative Bus `Azure Service Bus Messaging` as a plan B.
+4. Contribute both implementations to `Knative Eventing`.
+
+## Status
+
+Proposed
```
Comment (klaudiagrz) [inline architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md]: Please add the date here.

**Pull request merged**